File: app/models/interaction.js
import DS from 'ember-data';
import attr from 'ember-data/attr';

export default DS.Model.extend({
    action: attr('string'),
    extra_info: attr('string'),
    current_url: attr('string'),
    browser_info: attr('string'),
    session_info: attr('string')
});

File: backend/models/interactions/interaction.py
from sqlalchemy.dialects.postgresql.json import JSONB

from backend import db
from backend.models import BaseColumnsMixin, DictMixin


class Interaction(db.Model, BaseColumnsMixin, DictMixin):
    __tablename__ = 'user_interaction'

    action = db.Column(db.String(), default="watched")
    extra_info = db.Column(JSONB(astext_type=db.Text()), nullable=True)
    current_url = db.Column(db.Text())
    browser_info = db.Column(JSONB(astext_type=db.Text()), nullable=False)
    session_info = db.Column(JSONB(astext_type=db.Text()))
